src/atoms/referencesState.ts

```diff
@@ -0,0 +1,31 @@
+import { atom } from 'jotai';
+
+import { ReferenceItem } from '../types/ReferenceItem';
+
+const pdfIngestionProcessAtom = atom<string[]>([]); // ids of pending processes for now, this could be a record with progress percentage in the future
+
+export const isPdfIngestionRunningAtom = atom((get) => get(pdfIngestionProcessAtom).length > 0);
+
+export const addPdfIngestionProcessAtom = atom(null, (_get, set, value: string) => {
+  set(pdfIngestionProcessAtom, (pendingProcesses) => [...pendingProcesses, value]);
+});
+
+export const removePdfIngestionProcessAtom = atom(null, (_get, set, value: string) => {
+  set(pdfIngestionProcessAtom, (pendingProcesses) => pendingProcesses.filter((processId) => processId !== value));
+});
```

Comment:
Same for the removal, shouldn't one use only `pdfIngestionProcessAtom`

Reply:
Same as before, the idea was to decouple the atom to say that a process has ended, from the actual representation of processes.